chain/chain/src/update_shard.rs

```diff
@@ -106,15 +131,25 @@ pub(crate) fn process_shard_update(
     epoch_manager: &dyn EpochManagerAdapter,
     shard_update_reason: ShardUpdateReason,
     shard_context: ShardContext,
-    state_patch: SandboxStatePatch,
-) -> Result<ApplyChunkResult, Error> {
+    storage_context: StorageContext,
```

Comment:
nit: I would move storage_context to NewChunkData and OldChunkData since it is only needed there